conda_build/variants.py

```diff
@@ -533,22 +533,31 @@ def get_vars(variants, loop_only=False):
 @memoized
 def find_used_variables_in_text(variant, recipe_text):
     used_variables = set()
+    recipe_lines = recipe_text.splitlines()
     for v in variant:
-        variant_regex = r"\{\s*(?:pin_[a-z]+\(\s*?['\"])?%s[^'\"]*?\}\}" % v
-        selector_regex = r"^[^#\[]*?\#?\s\[[^\]]*?(?<![_\w\d])%s[=\s<>!\]]" % v
-        conditional_regex = r"[^\{]*?\{%\s*(?:el)?if\s*" + v + r"\s*(?:[^%]*?)?%\}"
-        # plain req name, no version spec.  Look for end of line after name, or comment or selector
-        requirement_regex = r"^\s+\-\s+%s(?:\s+[\[#]|$)" % v.replace('_', '[-_]')
-        all_res = [variant_regex, selector_regex, conditional_regex, requirement_regex]
+        all_res = []
         compiler_match = re.match(r'(.*?)_compiler$', v)
         if compiler_match:
+            compiler_lang = compiler_match.group(1)
             compiler_regex = (
-                r"\{\s*compiler\([\'\"]%s[\"\'][^\{]*?\}" % compiler_match.group(1)
+                r"\{\s*compiler\([\'\"]%s[\"\'][^\{]*?\}" % re.escape(compiler_lang)
             )
             all_res.append(compiler_regex)
+            variant_lines = [line for line in recipe_lines if v in line or compiler_lang in line]
+        else:
+            variant_lines = [line for line in recipe_lines if v in line]
```

Comment:
This misses lines where variant's `_` is replaced with `[-_]` in Line 556

Reply:
Dang it! Yeah, I added that preselection afterwards and apparently wasn't careful enough with that; the tests caught the `compiler` special handling but not the `[-_]` one.
I wonder how much performance impact this change alone has and whether it makes sense to keep it at all or dump it again to reduce code complexity...
I'll do some benchmarking on the conda-forge feedstocks' `meta.yaml`s and see how to add tests for this.

Reply:
Rendering all conda-forge feedstocks (with `finalize=False, bypass_env_check=True`) is 15-20 % faster on average with that extra step -- so, I guess it makes sense to keep it.